pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java

```diff
@@ -172,8 +172,17 @@ protected ConsumerBase(PulsarClientImpl client, String topic, ConsumerConfigurat
     @Override
     public Message<T> receive(int timeout, TimeUnit unit) throws PulsarClientException {
         if (conf.getReceiverQueueSize() == 0) {
-            throw new PulsarClientException.InvalidConfigurationException(
-                    "Can't use receive with timeout, if the queue size is 0");
+
+            if(!(getState() == State.Connecting)){
```

Comment:
Thanks for your contribution.
But I think this is not a right way to fix this.

Reply:
I think this way. The meaning of this state is whether the connection between each consumer and the topic is alive. If the task is cancelled or interrupted during the process of receiving the message, the downstream consumer should be notified in time

Reply:
We already check the state of consumer by `verifyConsumerState`.

Reply:
Good two, this function has been refactored, can you take time to have a look？
@linlinnn @zymap